### PR TITLE
GDB-4769 Nesting triples

### PR DESCRIPTION
**Pull request by svilenvelikov**
* Implement logic to mark nesting groups
* Styling the mapping groups
* Implement test for the nesting

https://ontotext.atlassian.net/browse/GDB-4769

### CODE REVIEW

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 1]
```diff
@@ -181,31 +188,44 @@ export class IterationComponent extends OnDestroyMixin implements OnInit, AfterV
...
+      propertyMappings.forEach((property) => {
+        this.setUsedSources(property);
+        if (property.getValues()) {
+          property.getValues().forEach((object) => {
+            this.setUsedSources(object);
+            this.addTriple(new Triple(subject, property, object, false, isRoot, isIRI), nestingLevel);
+            isRoot = false;
+            if (object.getValueType() && object.getValueType().getType() === Type.IRI) {
+              this.setTypeMappings(object, false, true);
+              nestingLevel++;
```
Comment (desislava-hristova-ontotext) [inline src/app/main/mapper/iteration/iteration.component.ts:215]: This increases the nestingLevel for each property + object, but all properties that are children if the same object IRI must be indented in the same level. The level must be increased only once.

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 2]
```diff
@@ -181,31 +188,44 @@ export class IterationComponent extends OnDestroyMixin implements OnInit, AfterV
...
-    this.getTypeMappings(subject) && this.getTypeMappings(subject).forEach((mapping) => {
-      this.setUsedSources(mapping);
-      this.triples.push(new Triple(subject, undefined, mapping, true, isRoot, isIRI));
-      isRoot = false;
-    });
+    const typeMappings = this.getTypeMappings(subject);
+    if (typeMappings) {
+      typeMappings.forEach((mapping) => {
+        this.setUsedSources(mapping);
+        this.addTriple(new Triple(subject, undefined, mapping, true, isRoot, isIRI), 0);
```
Comment (desislava-hristova-ontotext) [inline src/app/main/mapper/iteration/iteration.component.ts:195]: Type mappings are not always on level 0. If the type mapping has a subject that is the object of a previous triple, its level is the previous triple level + 1.

File: src/app/main/mapper/iteration/iteration.component.ts (hunk 1 above)
Comment (svilenvelikov) [reply src/app/main/mapper/iteration/iteration.component.ts:215]: I think this does just that. All properties which has the same object as a subject are assigned one and the same level.

File: src/app/main/mapper/iteration/iteration.component.ts (hunk 2 above)
Comment (svilenvelikov) [reply src/app/main/mapper/iteration/iteration.component.ts:195]: I don't get this. Can you give an example for such mapping or can we dicuss this in a call?

**Pull request merged**